src/vsr/message_header.zig

```diff
@@ -167,6 +167,7 @@ pub const Header = extern struct {
         if (self.checksum_body_padding != 0) return "checksum_body_padding != 0";
         if (self.nonce_reserved != 0) return "nonce_reserved != 0";
         if (self.size < @sizeOf(Header)) return "size < @sizeOf(Header)";
+        if (self.size > constants.message_size_max) return "size > message_size_max";
```

Comment:
:heart:

Reply:
I've was looking at this code today, and it seems that it might be beneficial to straight up assert here? 

checksum&header define framing, so they _must_ be validated by the caller. MessageBus validates them:

```
                    if (header.size < @sizeOf(Header) or header.size > constants.message_size_max) {
                        log.err("header with invalid size {d} received from peer {}", .{
                            header.size,
                            connection.peer,
                        });
                        connection.terminate(bus, .shutdown);
                        return null;
                    }
```

but this code isn't vopered, so it would be great to assert that on the replica's side for defense in depth.

Reply:
> it seems that it might be beneficial to straight up assert here?

Maybe it's too intrusive ... For example, AOF recovery and other tools may need to deal with malformed messages, so `header.invalid()` should not panic, and the `replica` instead calls `assert(header.invalid() == null)` when it needs to enforce the invariant.